File: benchmark/corpus/01_file_write/d_three_layers.py
# The write sits three calls down: main -> report -> render -> store.
import sys


def store(path, text):
    open(path, "w").write(text)  # DANGER


def render(path, total):
    text = "total %d" % total
    store(path, text)
    return text


def report(path, amounts):
    return render(path, sum(amounts))


path = sys.stdin.readline().strip()
print(report(path, [2500, 45000, 12000]))

File: benchmark/corpus/01_file_write/d_three_layers.js
// The write sits three calls down: main -> report -> render -> store.
function store(path, text) {
  Deno.writeTextFileSync(path, text); // DANGER
}

function render(path, total) {
  const text = "total " + total;
  store(path, text);
  return text;
}

function report(path, amounts) {
  return render(path, amounts.reduce((a, b) => a + b, 0));
}

const path = (await new Response(Deno.stdin.readable).text()).trim();
console.log(report(path, [2500, 45000, 12000]));
